apps/mn-synt-arbeidsforhold-service/src/main/java/no/nav/registre/testnorge/mn/syntarbeidsforholdservice/service/ArbeidsforholdHistorikkService.java

```diff
@@ -0,0 +1,179 @@
+        return list;
+    }
+
+    public void reportAll(LocalDate fom, LocalDate tom, int maxIdenter, String miljo) {
+
+        var startTimeStamp = LocalDateTime.now();
+
+        log.info("Starter syntentisering ({}).", startTimeStamp);
+        var dates = findAllDatesBetween(fom, tom);
+        var identer = identService.getIdenterUtenArbeidsforhold(miljo, maxIdenter);
+        if (identer.isEmpty()) {
+            log.warn("Fant ingen identer avsluter syntentisering...");
+        }
+
+        log.info("Syntentiser for {} person(er) mellom {} - {}...", identer.size(), fom, tom);
+        var arbeidsforholdMapList = getArbeidsforholdMapList(identer, dates);
+
+        for (var kalenermnd : dates) {
+            report(arbeidsforholdMapList, kalenermnd, miljo);
+        }
+        log.info("Syntentisering ferdig for {}/{} person(er) ({} - {}).",
+                arbeidsforholdMapList.size(),
+                identer.size(),
+                startTimeStamp,
+                LocalDateTime.now()
+        );
+    }
+
+    private void report(List<ArbeidsforholdMap> arbeidsforholdMapList, LocalDate kalenermnd, String miljo) {
+        var opplysningspliktige = opplysningspliktigService.getAllOpplysningspliktig(kalenermnd, miljo);
+        for (var arbeidsforholdMap : arbeidsforholdMapList) {
+            var arbeidsforhold = arbeidsforholdMap.getArbeidsforhold(kalenermnd);
+            log.info("Legger til arbeidsforhold for {} den {}.", arbeidsforhold.getIdent(), kalenermnd);
+            if (arbeidsforholdMap.isNewArbeidsforhold(kalenermnd)) {
+                var opplysningspliktig = opplysningspliktige.get(random.nextInt(opplysningspliktige.size()));
+                arbeidsforhold.setVirksomhetsnummer(opplysningspliktig.getRandomVirksomhetsnummer());
+                log.info("{} starter nytt arbeidsfohold den {}.", arbeidsforhold.getIdent(), kalenermnd);
+            } else {
+                var virksomhetsnummer = arbeidsforholdMap.getArbeidsforhold(kalenermnd.minusMonths(1)).getVirksomhetsnummer();
+                if (virksomhetsnummer == null) {
+                    throw new RuntimeException("Finner ikke forige virksomhentsnummer");
+                }
+                arbeidsforhold.setVirksomhetsnummer(virksomhetsnummer);
+            }
+            var opplysningspliktig = findOpplysningspliktigForVirksomhent(arbeidsforhold.getVirksomhetsnummer(), opplysningspliktige);
+            opplysningspliktig.addArbeidsforhold(arbeidsforhold);
+        }
+        opplysningspliktigService.send(opplysningspliktige, miljo);
+    }
+
+
+    private Opplysningspliktig findOpplysningspliktigForVirksomhent(String virksomhentsnummer, List<Opplysningspliktig> list) {
+        return list.stream()
+                .filter(value -> value.driverVirksomhent(virksomhentsnummer))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Finner ikke opplysningspliktig for virksomhentsnummer: " + virksomhentsnummer));
```

Comment:
"Finner ikke opplysningspliktig for virksomhetsnummer: "
Og endre variabel til virksomhetsnummer 😄